src/cli/java/org/commcare/util/screen/QueryScreen.java

```diff
@@ -143,15 +156,17 @@ public void answerPrompts(Hashtable<String, String> answers) {
 
             // If select question, we should have got an index as the answer which should
             // be converted to the corresponding value
-            if (queryPrompt.isSelectOne() && !answer.isEmpty()) {
-                int choiceIndex = Integer.parseInt(answer);
+            if (queryPrompt.isSelect() && !answer.isEmpty()) {
+                String[] selectedChoices = RemoteQuerySessionManager.extractSelectChoices(answer);
                 Vector<SelectChoice> selectChoices = queryPrompt.getItemsetBinding().getChoices();
-                if (choiceIndex < selectChoices.size()) {
-                    answer = selectChoices.get(choiceIndex).getValue();
-                } else {
-                    // answer is no longer a valid choice, so clear it out
-                    answer = "";
+                StringBuilder answerWithSelectValues = new StringBuilder();
+                for (int i = 0; i < selectedChoices.length; i++) {
+                    int choiceIndex = Integer.parseInt(selectedChoices[i]);
+                    boolean validChoice = choiceIndex < selectChoices.size() && choiceIndex > -1;
+                    answerWithSelectValues.append(validChoice ? selectChoices.get(choiceIndex).getValue() : "");
+                    answerWithSelectValues.append(i == selectedChoices.length - 1 ? "" : RemoteQuerySessionManager.MULTI_SELECT_DELIMITER);
```

Comment:
similar comment to on https://github.com/dimagi/formplayer/pull/973/: might be cleaner to put the indices in an array or list and then use `String.join` with the delimiter to get the final result.

Reply:
addressed.